scripts/create-release-changelog.mjs

```diff
@@ -0,0 +1,136 @@
+import * as child from 'child_process';
+import {
+  writeFileSync,
+  existsSync,
+  mkdirSync,
+} from 'fs';
+
+const githubLink = 'https://github.com/infermedica/component-library/pull/';
+const releaseDate = new Date().toLocaleString('en-US', {
+  year: 'numeric',
+  month: 'long',
+  day: 'numeric',
+});
+
+const changelogSections = {
+  feat: {
+    icon: '🚀',
+    name: 'Feature',
+  },
+  fix: {
+    icon: '🐛',
+    name: 'Fixes',
+  },
+  docs: {
+    icon: '📄',
+    name: 'Docs',
+  },
+  chore: {
+    icon: '🧹',
+    name: 'Chores',
+  },
+  refactor: {
+    icon: '📝',
+    name: 'Refactors',
+  },
+  breakingChanges: {
+    icon: '⛔️',
+    name: 'Breaking Changes',
+  },
+  test: {
+    icon: '🔍',
+    name: 'Tests',
+  },
+};
+
+const saveFile = (tag, content) => {
+  const [
+    major,
+    minor,
+  ] = tag.split('.');
+  const releaseVersionTag = `v${tag}`;
+  const releaseDirPath = `./docs/releases/v${major}.${minor}.x/${releaseVersionTag}`;
+
+  if (!existsSync(releaseDirPath)) {
+    mkdirSync(releaseDirPath);
+  }
+
+  try {
+    writeFileSync(`${releaseDirPath}/changelog.stories.mdx.mdx`, content[0]);
+    console.log(`🚀 Release ${releaseVersionTag} file has been created!`);
+  } catch (err) {
+    console.error(`⛔️ Something goes wrong and the Release ${releaseVersionTag} file hasn't been created!`, err);
+  }
+};
+
+const createChangelogSection = (content) => {
+  const commits = content.split(/\n/);
+  const sections = { breakingChanges: [] };
+  const prefixRegex = /^(.*?):/;
+  let sectionsDoc = '';
+
+  const updateChangelogSections = commits.reduce((acc, currentCommit) => {
+    let commitPrefix = currentCommit.match(prefixRegex) && currentCommit.match(prefixRegex)[1];
+    if (commitPrefix) commitPrefix = commitPrefix.replace(/\(.*?\)/, '');
+
+    if (commitPrefix in sections) {
+      sections[commitPrefix].push(currentCommit);
+    } else if (commitPrefix.search(/!/) > -1) {
+      sections.breakingChanges.push(currentCommit);
+    } else {
+      sections[commitPrefix] = [];
+      sections[commitPrefix].push(currentCommit);
+    }
+    return sections;
+  }, sections);
+
+  Object.keys(updateChangelogSections).reduce((acc, section) => {
```

Comment:
performance alert! `reduce` return new object like `map` etc.

Reply:
are we need this part of script? we can replace things in upper .reduce; add some `formatCommit(currentCommit)` function

Reply:
maybe you can try to use [Regex groups with .replace() in Javascript](https://www.codemzy.com/blog/regex-groups-with-replace) to replace `tag` to empty and handle PR hash